gpytorch/gpytorchopt/gpytorchopt/acquisition/functional/acquisition.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+from gpytorch import Module
+import torch
+from torch.distributions import Normal
+
+
+def expected_improvement(
+    model: Module,
+    candidate_set: torch.Tensor,
+    f_best: float,
+) -> torch.Tensor:
+    """
+    Parallel evaluation of single-point expected improvement, assumes
+    maximization
+
+    Args:
+        model (Module): GP from gpytorch
+        candidate_set (torch.Tensor): 'r x n' dimensional tensor where r is the number of
+        points to evaluate, and n is the number of parameters
+        fbest: previous best value
+    """
+    model.eval()
+    model.likelihood.eval()
+
+    pred = model.likelihood(model(candidate_set))
+
+    mu = pred.mean().detach()
+    sigma = pred.std().detach()
+
+    u = (f_best - mu) / sigma
+    m = Normal(torch.Tensor([0.0]), torch.Tensor([1.0]))
+    ucdf = m.cdf(u)
+    updf = torch.exp(m.log_prob(u))
+    ei = sigma * (updf + u * ucdf)
+
+    return ei
+
+
+def posterior_mean(
+    model: Module,
+    candidate_set: torch.Tensor,
+) -> torch.Tensor:
+    """
+    Parallel evaluation of single-point posterior mean
+
+    Args:
+        model (Module): GP from gpytorch
+        candidate_set (torch.Tensor): 'r x n' dimensional tensor where r is the number of
+        points to evaluate, and n is the number of parameters
+    """
+    pred_rv = model(candidate_set)
+    mean = pred_rv.mean()
+    return mean
+
+
+def probability_of_improvement(
+    model: Module,
+    candidate_set: torch.Tensor,
+    fbest: float,
+) -> torch.Tensor:
+    """
+    Parallel evaluation of single-point probability of improvement, assumes
+    maximization
+
+    Args:
+        model (Module): GP from gpytorch,
+        candidate_set (torch.Tensor): 'r x n' dimensional tensor where r is the number of
+        points to evaluate, and n is the number of parameters,
+        fbest: previous best point
+    """
+    normal = torch.distributions.Normal(0, 1)
+    pred_rv = model(candidate_set)
+    mean = pred_rv.mean()
+    sd = pred_rv.covar().diag().sqrt()
+    # TODO: ensure works with sd = 0.0
+    u = (mean - fbest) / sd
+    return normal.cdf(u)
+
+
+def upper_confidence_bound(
+    model: Module,
+    candidate_set: torch.Tensor,
+    beta: float,
+) -> torch.Tensor:
+    """
+    Parallel evaluation of single-point UCB, assumes maximization
+
+    Args:
+        model (Module): GP from gpytorch
+        candidate_set (torch.Tensor): 'r x n' dimensional tensor where r is the number of
+        points to evaluate, and n is the number of parameters
+        beta: used to trade-off mean versus covariance
+    """
+    pred_rv = model(candidate_set)
+    mean = pred_rv.mean()
+    var = pred_rv.covar().diag() * beta
+    return mean + var.sqrt()
+
+
+def max_value_entropy_search(
+    model: Module,
+    candidate_set: torch.Tensor,
+    num_samples: int,
+) -> torch.Tensor:
+    model.eval()
```

Comment:
Looks like in the other functions you expect the model to already be in eval mode? We should be consistent across functions.